base/compiler/abstractinterpretation.jl

```diff
@@ -1853,44 +1853,46 @@ function abstract_call_builtin(interp::AbstractInterpreter, f::Builtin, (; fargs
 end
 
 function abstract_call_unionall(interp::AbstractInterpreter, argtypes::Vector{Any}, call::CallMeta)
-    if length(argtypes) == 3
-        canconst = true
+    na = length(argtypes)
+    if isvarargtype(argtypes[end])
+        na == 3 || return CallMeta(Any, EFFECTS_THROWS, call.info)
```

Comment:
na==4 is equally safe to infer as 3. Make than 4 and it will always throw

Reply:
Right, thanks for your comment.